pkg/rke2/rke2.go

```diff
@@ -53,27 +52,8 @@ func Agent(clx *cli.Context, cfg Config) error {
 }
 
 func setup(clx *cli.Context, cfg Config) error {
-	var dataDir string
-
-	for _, f := range clx.Command.Flags {
-		switch t := f.(type) {
-		case cli.StringFlag:
-			if strings.Contains(t.Name, "data-dir") {
-				dataDir = *t.Destination
-			}
-		}
-	}
-
-	for _, f := range clx.App.Flags {
-		switch t := f.(type) {
-		case cli.StringFlag:
-			if t.Name == "profile" && t.Destination != nil && *t.Destination != "" {
-				cisMode = true
-			}
-		default:
-			// nothing to do. Keep moving.
-		}
-	}
+	cisMode = clx.String("profile") != ""
+	dataDir := clx.String("data-dir")
```

Comment:
I have to ask why? 
Why did we iterate through the flags before and why do we not need to now.

Reply:
I suspect this was either necessary when using a different CLI library, or sometime didn't know the right way to do it.